docs/developer-docs/production/resource-limits.md

```diff
@@ -0,0 +1,46 @@
+# Resource Limits
+
+## Overview
+
+The Internet Computer uses WebAssembly as the platform for executing messages of smart contracts.
+Since WebAssembly is [Turing-complete](https://en.wikipedia.org/wiki/Turing_completeness), it can express different kinds of computations including non-terminating computations.
+The Internet Computer protects against non-terminating computations by limiting the number of WebAssembly instructions and other resources per message execution.
+The limits depend on the message type as shown in the following table.
+
+## Resource constraints and limits
+
+| Resource                                                  | Constraint                |
+| --------------------------------------------------------- | ------------------------- |
+| Canister queue limit                                      | 500 messages              |
+| Maximum ingress and cross-net inter-canister call payload | 2MB                       |
+| Maximum same-subnet inter-canister call payload¹          | 10MB¹                     |
+| Maximum response size                                     | 2MB                       |
+| Instruction limit per update call/heartbeat/timer         | 20B per method invocation |
+| Instruction limit query calls                             | 5B                        |
+| Instruction limit for canister install and upgrade        | 200B                      |
+| Subnet capacity                                           | 700GB                     |
+| Wasm heap size                                            | 4GB                       |
+| Wasm stable memory                                        | 96GB                      |
+| Wasm custom sections per subnet                           | 2GB                       |
+| Wasm custom sections per canister                         | 1MB                       |
+| Wasm custom sections per canister                         | 16 sections at most       |
+| Wasm code section                                         | 10MB                      |
+| Query calls execution threads                             | 2 per replica node        |
+| Update calls execution threads                            | 4 per subnet              |
+
+Notes:
+
+1. The limit may be deprecated at some point.
```

Comment:
What does this refer to?

Reply:
I moved the note back to the table, PTAL.

Reply:
LGTM! Thanks!